### ISSUE TITLE
Deprecate io.ana code?

### ISSUE THREAD

**Issue by dstansby** [sunpy/sunpy#4412]
`sunpy.io.ana` contains wrappers for C code to read in .ana files. It is suspected that these files are not widely used, and therefore it would be good from a maintenance and "making sunpy pure python" point of view to deprecate and eventually remove this code.

A deprecation message should
- State that the sub-module is planned for removal in 4.0 (this gives us 1 LTS cycle)
- Give clear contact details (probably https://groups.google.com/forum/#!forum/sunpy) to contact if you are using the code and would like to see it remain

**Comment by Cadair** [comment]
I think this is worth a little discussion. While I agree there is a complexity increase in us having a compiled extension, most of the end user issues around this have been mitigated by changes to the python packaging ecosystem in the last few years.

The C code is far from perfect, and we don't have a lot of expertise in maintaining it, however, we also basically never have to touch it. The only time it got any serious attention was the python 3 upgrade, and even that wasn't hard. What are we actually gaining by removing this?!

I would be interested in knowing if anyone actually uses it though, we should probably email out to the mailing list to see if anyone on that list is using it. I would rather see us attempt to reach out to people first, rather than "tell us if you don't want us to break your shit".

**Comment by ayshih** [comment]
I think that having a goal of "pure Python" is a bit silly.  I'd like to see more C and Cython code in SunPy!

However, the ANA code in specific is problematic for reasons that are related to the fact that it is C code, at least not directly.  I believe that it's still the case the ANA code doesn't give the same output on Windows as on other platforms (https://github.com/sunpy/sunpy/issues/745#issuecomment-580329561).  Presumably the code is wrong on Windows, but it needs someone with familiarity to debug.  (Many years ago I believe I verified the output at the time by cross-checking against ANA code in SSW, but I do not guarantee that I did that correctly.)
